### PR TITLE
Retry popen on 127 osx

### PR DESCRIPTION
**Pull request by autoantwort**
Close https://github.com/microsoft/vcpkg/issues/21905 bullet point 1.  

Code 127 normally means `command not found`, but that does not really makes sense here. Sometimes (not often) `unzip` returns 127 and sometimes not, when you retry it the return code is 0. I have no idea what the problem could be, but this implements a workaround by retrying the command once on mac when the return code is 127.

### CODE REVIEW

File: src/vcpkg/base/system.process.cpp [hunk 1]
```diff
@@ -773,7 +774,11 @@ namespace vcpkg
                                     const Environment& env)
     {
         const auto timer = ElapsedTimer::create_started();
-
+        const auto thread_id = []() {
+            std::ostringstream ss;
+            ss << std::this_thread::get_id();
+            return ss.str();
+        }();
```
Comment (strega-nil-ms) [inline src/vcpkg/base/system.process.cpp:781]: This is not a very good way to do this...

File: src/vcpkg/base/system.process.cpp [hunk 2]
```diff
@@ -803,32 +808,59 @@ namespace vcpkg
...
+                exit_code = WTERMSIG(exit_code);
+            }
+            else if (WIFSTOPPED(exit_code))
+            {
+                exit_code = WSTOPSIG(exit_code);
+            }
+#ifdef __APPLE__
+            if (exit_code == 127 && !got_output)
+            {
+                Debug::print(thread_id, ": pclose returned 127, try again \n");
```
Comment (strega-nil-ms) [inline src/vcpkg/base/system.process.cpp:853]:
```suggestion
                Debug::print(std::to_string(std::this_thread::get_id()), ": pclose returned 127, try again \n");
```

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]
There are some clear (minor) changes needed, but I want someone else to take a look and see if we want to do this.

**Review by strega-nil-ms — APPROVED** [review]
nevermind; I am so annoyed that this is the best way to get a string representation of a thread ID.

**Comment by ras0219-msft** [comment]
I merged https://github.com/microsoft/vcpkg-tool/pull/305 since it seems obviously positive. I am less convinced about arbitrarily retrying: child processes are not guaranteed to be idempotent.

If we want to implement a retry around "unzip" specifically on mac, I think that would be reasonable. It should not be implemented for _all_ processes though.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
See above comment.

**Review by ras0219-msft — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks, LGTM!

**Pull request merged**